operator/watchers/pod.go

```diff
@@ -24,18 +26,11 @@ var (
 	// operations in k8s as some of its fields are not populated.
 	PodStore cache.Store
 
-	// PodStoreSynced is closed once the PodStore is synced with k8s.
-	PodStoreSynced = make(chan struct{})
-
 	// UnmanagedKubeDNSPodStore has a minimal copy of the unmanaged kube-dns pods running
 	// in the cluster.
 	// Warning: The pods stored in the cache are not intended to be used for Update
 	// operations in k8s as some of its fields are not populated.
 	UnmanagedKubeDNSPodStore cache.Store
-
-	// UnmanagedPodStoreSynced is closed once the UnmanagedKubeDNSPodStore is synced
```

Comment:
Heh, this channel was the cause of the original flaky test and it wasn't even used anywhere (same for PodStoreSynced). The changes in this PR still make sense since there's still e.g. `slimNodeStoreSynced`.